app/src/ui/branches/branches-container.tsx

```diff
@@ -397,11 +394,18 @@ export class BranchesContainer extends React.Component<
       return
     }
 
-    if (this.props.isCherryPickInProgress) {
+    if (dragAndDropManager.isDragOfType(DragType.Commit)) {
       this.props.dispatcher.startCherryPickWithBranch(
         this.props.repository,
         branch
       )
     }
   }
+
+  private onDropOntoCurrentBranch = () => {
+    if (dragAndDropManager.isDragOfType(DragType.Commit)) {
+      this.props.dispatcher.endCherryPickFlow(this.props.repository)
+      this.props.dispatcher.recordCherryPickDragStartedAndCanceled()
```

Comment:
Hmm.. note for later.. this statistics name should be refactored to just `recordDragStartedAndCanceled`